app/frontend/actions.ts

```diff
@@ -860,7 +614,7 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
       const validationError =
         delegationPlanValidator(balance, 0 as Lovelace, txPlanResult.estimatedFee) ||
         txPlanResult.error
-      setErrorState('delegationValidationError', validationError, {
+      setErrorState(state, 'delegationValidationError', validationError, {
```

Comment:
General concern: are we sure the state you are passing here is always up to date?